log.go

```diff
@@ -126,12 +127,24 @@ func S() *zap.SugaredLogger {
 	return globalSugarLogger.Load().(*zap.SugaredLogger)
 }
 
-// ReplaceGlobals replaces the global Logger and SugaredLogger.
-// It's safe for concurrent use.
-func ReplaceGlobals(logger *zap.Logger, props *ZapProperties) {
+// ReplaceGlobals replaces the global Logger and SugaredLogger, and returns a
+// function to restore the original values. It's safe for concurrent use.
+func ReplaceGlobals(logger *zap.Logger, props *ZapProperties) func() {
```

Comment:
Do we regard `ReplaceGlobals` as public interface? This change affects signature and will create significant backward compatibility breaker.

Reply:
The current `ReplaceGlobals` does not have a return value, so that it is used as:

```go
log.ReplaceGlobals(...)
```

As you can see, adding a return value (which does not need error checks) doesn't break these codes after upgrading.

Reply:
Yes for the case. However, if someone passes it as a parameter, the signature changed are significant.

Anyway, if we think pingcap/log as an internal package and only promise its usage as we image, you can make the change.

Reply:
I'm not sure about whether pingcap/log is internal or not 🤔 Maybe we can bump semantic version to make a breaking change?

Reply:
That sounds reasonable to me.

Reply:
Oh. I just discovered that we are even not using any version for this library 💩 ....  Then how about just making this change?

Reply:
That makes sense. I'd prefer we add a version but it should not be a blocker to this one. Could you create an issue for it? I'm resolving this conversation since we don't push it implemented in this PR.